lib/src/code_generation/locator/stacked_locator_annotations.dart

```diff
@@ -120,17 +120,33 @@ const factoryParam = FactoryParam._();
 /// Registers the type passed in to be presolved using the function passed in
 class Presolve extends DependencyRegistration {
```

Comment:
Let's add a deprecation notice here. We'll publish this once the docs are ready with a link to the docs